public/js/validators/user.js

```diff
@@ -0,0 +1,56 @@
+/** @fileoverview This file will contain all the required validations for signup and login */
+
+import { validationsForStrings, isNumber } from "./util.js";
+
+/**
+ * This function will validate all the fields required for signup process
+ * 
+ * @param {{ 
+ * 	firstName: String, 
+ * 	lastName: String, 
+ * 	email: String, 
+ * 	age: Number, 
+ *  username: String, 
+ *  password: String 
+ * }} user 
+ * 
```

Comment:
These type of comments are called as jsdocs and they are helpful for documenting our code and also helps VSCode intellisense to what type of arguments a function is expecting, what is it's return value etc.
![image](https://user-images.githubusercontent.com/40670894/232181207-8bb94c47-5e43-4f85-bb20-51443ca13030.png)